sw/device/silicon_creator/lib/drivers/pinmux.c

```diff
@@ -126,10 +126,10 @@ static void configure_output(pinmux_output_t output) {
 }
 
 void pinmux_init(void) {
-  uint32_t bootstrap_en =
-      otp_read32(OTP_CTRL_PARAM_OWNER_SW_CFG_ROM_BOOTSTRAP_EN_OFFSET);
-  if (launder32(bootstrap_en) == kHardenedBoolTrue) {
-    HARDENED_CHECK_EQ(bootstrap_en, kHardenedBoolTrue);
+  uint32_t bootstrap_dis =
+      otp_read32(OTP_CTRL_PARAM_OWNER_SW_CFG_ROM_BOOTSTRAP_DIS_OFFSET);
+  if (launder32(bootstrap_dis) != kHardenedBoolTrue) {
+    HARDENED_CHECK_NE(bootstrap_dis, kHardenedBoolTrue);
```

Comment:
The goal of `HARDENED_CHECK_NE` is to detect when an attacker is attempting to glitch a variable, right? Shouldn't this be `HARDENED_CHECK_EQ(bootstrap_dis, kHardenedBoolFalse)` in that case?

Also, do we need a corresponding `else` block that checks `HARDENED_CHECK_EQ(bootstrap_dis, True)`?

Reply:
> The goal of `HARDENED_CHECK_NE` is to detect when an attacker is attempting to glitch a variable, right? Shouldn't this be `HARDENED_CHECK_EQ(bootstrap_dis, kHardenedBoolFalse)` in that case?

Good question. Since the goal is to reduce risk, we decided to implement the following mapping from the set of 32-bit values to outcomes:
* `true` -> bootstrap disabled
* any other value -> bootstrap enabled.

instead of:

* `true` -> bootstrap disabled
* `false` -> bootstrap enabled
* any other value -> shutdown.

I know #15715 is a bit terse but that was the plan :)